db/hybrid_logical_vector.go

```diff
@@ -180,6 +183,32 @@ func (hlv *HybridLogicalVector) GetVersion(sourceID string) uint64 {
 	return latestVersion
 }
 
+// AddNewerVersions will take a hlv and add any newer source/version pairs found across CV and PV found in the other HLV taken as parameter
+func (hlv *HybridLogicalVector) AddNewerVersions(otherVector HybridLogicalVector) {
```

Comment:
Can we add some unit tests for AddNewerVersions to ensure we're covering all permutations?

In particular, I'm unsure that the current code handles cases like:
```
hlv:
   sourceID: abc,
   version: 15
otherVector:
   sourceID: def
   version: 25,
   PreviousVersions: "abc":20
expectedHLV:
   sourceID: def,
   version: 25,
   PreviousVersions: "abc":20
```
and
```
hlv:
   sourceID: abc,
   version: 15
   PreviousVersions:"def":30 
otherVector:
   sourceID: def
   version: 35,
   PreviousVersions: "abc":15
expectedHLV:
   sourceID: def,
   version: 35,
   PreviousVersions: "abc":15
```

Reply:
Good spot, exposeed some issue with my logic there. Added these test cases plus couple extra. 

There is some logic I have implemented in AddNewerVersions that accounts for otherVector having a lower CV. In this case AddVersion will error so added some code to add the CV from otherVector to PV in local HLV. Now in this specific use case I don't think this should ever be the case so I don't believe this code is needed but left it in as I feel it could be useful for some other scenarios where you may want to almost 'merge' two HLV structures? But if you disagree i'm happy to remove these liens of code. For reference I'm talking about lines 212 to 226 in AddNewerVersions.